### DISCUSSION TITLE
v3 ready: Tools and Plugins

### DISCUSSION THREAD

**Discussion by snorrees** [sanity-io/sanity#3343] [Announcements]
As we talked about during the v3 launch, here is the first batch of tools and plugin repositories that you can use in Sanity Studio v3 today! 

Take a look at the README in each repository to learn more about each plugin or tool.

**Note:** v3 plugins are released under the `studio-v3`  npm tag. 

You can install them using a regular npm command, for example:
```
npm install --save @sanity/color-input@studio-v3
```
then follow the usage section of each plugin.

`sanity install` is no longer used in Studio v3.

## Plugins
Lives in the sanity monorepo.
- [@sanity/vision](https://github.com/sanity-io/sanity/blob/v3/packages/@sanity/vision/README.md)

The following plugins used to live inside the [Sanity monorepo](https://github.com/sanity-io/sanity),
but have been extracted to standalone repositories now:

- [@sanity/color-input](https://github.com/sanity-io/color-input)
- [@sanity/code-input](https://github.com/sanity-io/code-input)
- [@sanity/google-maps-input](https://github.com/sanity-io/google-maps-input)
- [@sanity/dashboard](https://github.com/sanity-io/dashboard)
- [@sanity/language-filter](https://github.com/sanity-io/language-filter#sanitylanguage-filter)

The following plugins have now have a v3 version in their respective repos:

- [sanity-plugin-asset-source-unsplash](https://github.com/sanity-io/sanity-plugin-asset-source-unsplash/tree/v3)
- [sanity-plugin-markdown](https://github.com/sanity-io/sanity-plugin-markdown/tree/v3)
- [sanity-plugin-scheduled-publishing](https://github.com/sanity-io/sanity-plugin-scheduled-publishing/tree/v3)
- [sanity-plugin-graph-view](https://github.com/sanity-io/sanity-plugin-graph-view/tree/v3)
- [dashboard-widget-document-list](https://github.com/sanity-io/dashboard-widget-document-list/tree/v3) (widget for @sanity/dashboard)
- [example-dashboard-widget-cats](https://github.com/sanity-io/example-dashboard-widget-cats/tree/v3) (widget for @sanity/dashboard)
- [sanity-plugin-media](https://github.com/robinpyon/sanity-plugin-media/tree/v3)
- [sanity-plugin-latex-input](https://github.com/sanity-io/latex-input/tree/v3)
- [@sanity/document-internationalization](https://github.com/sanity-io/document-internationalization/tree/v3)
- [sanity-plugin-mux-input](https://github.com/sanity-io/sanity-plugin-mux-input)
- [orderable-document-list](https://github.com/sanity-io/orderable-document-list/tree/v3)
- [dashboard-widget-netlify](https://github.com/sanity-io/sanity-plugin-dashboard-widget-netlify/tree/v3)
- [dashboard-widget-vercel](https://github.com/sanity-io/sanity-plugin-dashboard-widget-vercel/tree/v3)
- [sanity-plugin-documents-pane](https://github.com/sanity-io/sanity-plugin-documents-pane/tree/v3)
- [@sanity/studio-secrets](https://github.com/sanity-io/sanity-studio-secrets/tree/v3) (called `sanity-secrets` in v2)
- [sanity-plugin-iframe-pane](https://github.com/sanity-io/sanity-plugin-iframe-pane/tree/studio-v3)
- [sanity-plugin-cloudinary](https://github.com/sanity-io/sanity-plugin-cloudinary/tree/v3)(v3 version includes `sanity-plugin-asset-source-cloudinary`)
- [sanity-plugin-hotspot-array](https://github.com/sanity-io/sanity-plugin-hotspot-array/tree/v3)
- [@sanity/table](https://github.com/sanity-io/table#sanity-table-plugin) (official V3 replacement for sanity-plugin-table)
- [sanity-plugin-bynder-input](https://github.com/sanity-io/sanity-plugin-bynder-input)

### Community made

- [Giphy asset source plugin](https://github.com/XSlemX/sanity-plugin-asset-source-giphy)

## Tools

- [@sanity/plugin-kit](https://github.com/sanity-io/plugin-kit)
- [@sanity/eslint-config-no-v2-imports](https://github.com/sanity-io/eslint-config-no-v2-imports)
- [@sanity/incompatible-plugin](https://github.com/sanity-io/incompatible-plugin)

Take specially note of.@sanity/plugin-kit, it provides CLI commands for create new, and updating existing plugins.

## The road ahead

Let us know which plugins you would like to see ported to Studio v3! 

Join in the fun and run `npx @sanity/plugin-kit verify-package` in your own Studio v2 plugin packages to get an idea of the steps needed to get there.

Create your very own v3 plugin package with `npx @sanity/plugin-kit init awesome-thing-i-made` and let us know about it!

**Comment by logemann** [comment]
Would love to see the following ones:

- dashboard-widget-netlify (but you are not the maintainer possibly)
- @sanity/google-maps-input
- @sanity/desk-tool

**Reply by snorrees** [reply]
I can give you 2 of 3 right now 😅 !

@sanity/desk-tool is already part of the V3 sanity package, and is a core part of configuring the studio. Take a look on this [V2 vs V3 config example](https://beta.sanity.io/docs/platform/studio/v2-to-v3/intro-to-migration#goodbye-parts!-hello-config-api!) in the new docs.

[@sanity/google-maps-input](https://github.com/sanity-io/google-maps-input) is V3 ready! It is nestled away in the list above :)

Taking note of dashboard-widget-netlify; since it is a dashboard widget is should be fairly straightforward to upgrade. 

[@sanity/dashboard](https://github.com/sanity-io/dashboard) got a couple of API changes in V3 as well, since parts are no longer a thing. Check out the "Upgrading from v2 bit" [here](https://github.com/sanity-io/dashboard#upgrading-from-v2).

**Reply by logemann** [reply]
awesome. Thanks and sorry that i overlooked the maps one in your list. Netlify dashboard widget is most likely used by a lot of users because its really kind of useful ;-)

**Reply by jkananen** [reply]
+1 to `dashboard-widget-netlify`

It seems I could already use V3 on the simplest studios if website deploys on Netlify were possible from Sanity dashboard with the plugin.

**Reply by jaydanurwin** [reply]
Same here on `dashboard-widget-netlify`. I could migrate all of my clients' Sanity desks to v3 if that was migrated.

**Reply by snorrees** [reply]
[dashboard-widget-netlify](https://github.com/sanity-io/sanity-plugin-dashboard-widget-netlify/tree/v3) is now out for v3.

**Reply by makerbrothersworldwide** [reply]
is it possible to do these as well?

https://www.sanity.io/plugins/social-preview

https://www.sanity.io/plugins/sanity-plugin-google-analytics

https://www.sanity.io/plugins/seo-pane

thanks!

**Comment by nickroberts** [comment]
Just starting a new project, and would love to see these:

* sanity-plugin-media (I know this one is coming soon!)
* sanity-plugin-cloudinary
* sanity-plugin-asset-source-cloudinary

I'm still trying to figure out which Cloudinary plugin (or vanilla Sanity media) would be best with our NextJs frontend.

Exciting times ahead! Awesome work!

**Reply by snorrees** [reply]
[sanity-plugin-media](https://github.com/robinpyon/sanity-plugin-media/tree/v3-studio) is out btw :)

**Reply by snorrees** [reply]
[@sanity/vison](https://github.com/sanity-io/sanity/blob/v3/packages/@sanity/vision/README.md) is out starting from dev-preview.13!

**Comment by babyPrince** [comment]
Mux :)
- [mux video input](https://www.sanity.io/plugins/sanity-plugin-mux-input)

_*crossing fingers for batch upload of video files in array of mux.input (drag and drop all together)_ 🤞

**Reply by stipsan** [reply]
The mux one is coming soon! That one is on me 😄

**Reply by babyPrince** [reply]
Wonderful :) Any possibilities for batch upload? hehe

By the way, would the plugin work on v2?

**Reply by stipsan** [reply]
It's [out](https://github.com/sanity-io/sanity-plugin-mux-input#quick-start):
```
npm i sanity-plugin-mux-input@studio-v3
```
> Any possibilities for batch upload?

Not in this release 🙂

**Comment by luigimannoni** [comment]
Definitely Vision, `sanity-plugin-media` with batch upload, `sanity-plugin-better-slug` for the v3, I do understand though the latter with the new form api should be easy to replicate in every project.

**Reply by snorrees** [reply]
[sanity-plugin-media](https://github.com/robinpyon/sanity-plugin-media/tree/v3-studio) is out!

**Reply by snorrees** [reply]
[@sanity/vison](https://github.com/sanity-io/sanity/blob/v3/packages/@sanity/vision/README.md) is out starting from dev-preview.13!

**Comment by valse** [comment]
Hi... with the `@sanity/document-internationalization` it would be useful the `@sanity/language-filter` plugin too ;)

**Reply by snorrees** [reply]
Yep, `@sanity/language-filter` with `@sanity/document-internationalization` makes a lot of sense, and is even recommended in the doc-18n README :)

`@sanity/language-filter` will definitely need a replacement, but I cant promise that we will have anything to replace it until a bit later on the dev-preview timeline. Watch this space ;)

**Reply by snorrees** [reply]
[@sanity/language-filter](https://github.com/sanity-io/language-filter#sanitylanguage-filter) is now available.

**Comment by AmeyGohil** [comment]
I installed the [@sanity/color-input](https://github.com/sanity-io/color-input) and followed the usage but in the `sanity.config.ts` file, it shows `Module '"@sanity/color-input"' has no exported member 'colorSchema'.ts(2305)` in the editor and giving the following error in the studio
```
Uncaught error: The requested module '/node_modules/.vite/deps/@sanity_color-input.js?v=344716ad' does not provide an export named 'colorSchema'
http://localhost:3333/sanity.config.ts?t=1655364542058:3:10
SyntaxError: The requested module '/node_modules/.vite/deps/@sanity_color-input.js?v=344716ad' does not provide an export named 'colorSchema'
```

**Reply by snorrees** [reply]
Oh, you found a typo in the documentation. Export was renamed to colorInput at the last moment, and I see the README was left behind.

It should  be:

```js
import { colorInput } from "@sanity/color-input";

export default createConfig({
  // ...
  plugins: [
    colorInput(),
  ] 
})
```

I have [released a new version](https://github.com/sanity-io/color-input#usage) with the text update; you dont have to update - fixing the import should work in the current version you have :)

Thanks for reporting!

**Comment by babyPrince** [comment]
One more very useful could be [Vercel Deploy](https://www.sanity.io/plugins/vercel-deploy) if possible

**Reply by snorrees** [reply]
Not the same plugin, but [dashboard-widget-vercel](https://github.com/sanity-io/sanity-plugin-dashboard-widget-vercel/tree/v3) is now out for v3, and should hopefully serve the same purpose.

**Comment by Gawdfrey** [comment]
I have updated the [Giphy asset source plugin](https://github.com/XSlemX/sanity-plugin-asset-source-giphy) to v3!

**Reply by kmelve** [reply]
That was quick! Awesome work and thank you!

**Reply by snorrees** [reply]
Fantastic! Would love to hear if the plugin-kit was of any help at all. 

I also see that we have linked to plugin-sdk instead of [plugin-kit](https://github.com/sanity-io/plugin-kit) in the last part of all our readmes 😅  . Another one of those "changed name at the last minute" things that is not checked by a compiler, and thus slipped through ;) I will update things in the released plugins over the weekend, you might want to do the same.

**Reply by Gawdfrey** [reply]
Thanks!

The plugin kit was definitely a great tool to have. Would have probably spent a few hours searching the docs and looking at example plugins to configure the v3 version without it. I have also started a fresh plugin repo with it now and i was up and running writing the plugin within minutes, which is amazing so great work on it so far!

I will also change the plugin-kit url, copy pastas are dangerous 😄

**Reply by evenwestvang** [reply]
So great to hear the kit was useful in porting! Thanks for doing this.

**Reply by XSlemX** [reply]
@Gawdfrey did such a great job porting/upgrading, that I finally woke up and reworked the UI.

**Reply by XSlemX** [reply]
@kmelve @evenwestvang the readme has been updated on the package, but the tools and plugin page is not.

**Comment by snorrees** [comment]
@valse [@sanity/document-internationalization](https://github.com/sanity-io/document-internationalization/tree/v3) is out! 

No language-filter quite yet: that bit is pending implementation in Studio core, so you will have to bear with us for now.

**Comment by edolyne** [comment]
Would love to see these updated for v3 as well

- [sanity-plugin-seo-pane](https://github.com/sanity-io/sanity-plugin-seo-pane)
- [sanity-plugin-hierarchical-document-list](https://github.com/sanity-io/hierarchical-document-list)
- [@sanity/orderable-document-list](https://github.com/sanity-io/orderable-document-list)

**Reply by snorrees** [reply]
[orderable-document-list](https://github.com/sanity-io/orderable-document-list/tree/v3) is out for v3 as of today.

**Reply by Designer023** [reply]
Seconding `seo-pane`

**Comment by bline** [comment]
Would really help me to have [@sanity/block-tools](https://github.com/sanity-io/sanity/tree/next/packages/@sanity/block-tools) for converting html to portable text outside of sanity studio.

**Reply by kmelve** [reply]
You can use block tools outside Sanity Studio today. Here's [an example](https://github.com/kmelve/markdown-to-sanity) of a project using it. It does require a schema, but it doesn't require the studio.

**Reply by bline108** [reply]
Right, I guess my blocker on that was the complications of schema custom inputs which require the parts system.

Really excited to see v3 solve these issues. Thanks!

**Comment by alex289** [comment]
Any updates for Vision or when we can expect it?

**Reply by snorrees** [reply]
[@sanity/vison](https://github.com/sanity-io/sanity/blob/v3/packages/@sanity/vision/README.md) is out starting from dev-preview.13!

**Reply by mehmet-karaca** [reply]
@codesandbox/sandpack-react does not work with

`Unhandled Runtime Error
Error: Unrecognized extension value in extension set ([object Object]). This sometimes happens because multiple instances of @codemirror/state are loaded, breaking instanceof checks.`

**Comment by ottob** [comment]
[sanity-plugin-markdown](https://github.com/sanity-io/sanity-plugin-markdown/tree/v3) works locally for me, but now when deployed. I have opened an issue here: https://github.com/sanity-io/sanity-plugin-markdown/issues/28

Does anyone else have the same problem?

**Reply by snorrees** [reply]
Thanks for the heads up! Released a new version that fixes the crash, but its a breaking change as we had to change the markdown editor to make it play nice with the build. 

https://github.com/sanity-io/sanity-plugin-markdown/releases/tag/v3.0.0-v3-studio.1

With this version you should have an working editor, but image upload is not working. Sorry!

**Comment by snorrees** [comment]
[@sanity/language-filter](https://github.com/sanity-io/language-filter#sanitylanguage-filter) now has a v3 version.

**Comment by mosesoak** [comment]
Hi, we currently use a tool called sanity-codegen https://github.com/ricokahler/sanity-codegen to generate types from our studio into our frontend. Invaluable!

Is that something that will be needed in v3, or will it have a way to export all of the types easily?

If it will, my guess is that it would need to be upgraded to work with v3, would that just fall to the original author? Thanks

**Comment by snorrees** [comment]
Vercel and Netlify dashboard widget plugins are now V3 ready:

- [dashboard-widget-netlify](https://github.com/sanity-io/sanity-plugin-dashboard-widget-netlify/tree/v3)
- [dashboard-widget-vercel](https://github.com/sanity-io/sanity-plugin-dashboard-widget-vercel/tree/v3)

**Comment by robinpyon** [comment]
[sanity-plugin-documents-pane](https://github.com/sanity-io/sanity-plugin-documents-pane/tree/v3) is now V3 ready

**Reply by Designer023** [reply]
Thanks  👍 It's a great plugin

**Comment by snorrees** [comment]
V3 ready:
- [@sanity/studio-secrets](https://github.com/sanity-io/sanity-studio-secrets/tree/v3) (called sanity-secrets in v2)
- [sanity-plugin-iframe-pane](https://github.com/sanity-io/sanity-plugin-iframe-pane/tree/studio-v3)
- [sanity-plugin-cloudinary](https://github.com/sanity-io/sanity-plugin-cloudinary/tree/v3)(v3 version includes sanity-plugin-asset-source-cloudinary)
- [sanity-plugin-hotspot-array](https://github.com/sanity-io/sanity-plugin-hotspot-array/tree/v3)

**Comment by ehuelsda** [comment]
Any plans for upgrading the [hierarchical-document-list](https://github.com/sanity-io/hierarchical-document-list) plugin?

**Reply by kmelve** [reply]
Hi! We're looking into this!

**Reply by ehuelsda** [reply]
@kmelve any progress on this? 😊

**Reply by ehuelsda** [reply]
@kmelve still waiting for this one. Can we get an update?

**Comment by snorrees** [comment]
[@sanity/table](https://github.com/sanity-io/table#sanity-table-plugin), V3 replacement for sanity-plugin-table is ready.

Big thanks to @davydog187 for doing all the migration work!

**Reply by davydog187** [reply]
@snorrees the plugin in the docs is showing the wrong contributor(s) (this was the fork) and the NPM / Github links are also wrong. https://www.sanity.io/plugins/table

Can you update?

**Comment by Adstokoe19** [comment]
Please could you make [next-auth-sanity](https://github.com/Fedeya/next-auth-sanity) V3 ready? Thanks.

**Reply by kmelve** [reply]
I think you have to ask [the maintainer for this plugin](https://github.com/Fedeya/next-auth-sanity/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc). That being said, I don't see any code in the plugin that wouldn't make it work with Studio v3 as it is now.

**Reply by Adstokoe19** [reply]
@kmelve Thanks. Yes the plugin works fine with v3. The schema requires updating (but still works!).
